File: Structs/struct.py
import time
from collections import namedtuple

# Definición de la estructura
Persona = namedtuple("Persona", ["nombre", "edad", "id"])

# Creación de la tupla/struct
personas = [Persona(nombre="Josue", edad=22, id=i) for i in range(2_000_000)]

# Medición de tiempo de acceso
start_time = time.time()
for persona in personas:
    _ = persona.nombre
    _ = persona.edad
    _ = persona.id

elapsed_time = (time.time() - start_time) * 1000
print("         /^\\/^\\")
print("       _|__|  O|")
print("\\/    /~     \\_/ \\")
print(" \\____|__________/  \\")
print("        \\_______      \\")
print("                `\\     \\                 \\")
print("                  |     |                  \\")
print("                 /      /                    \\")
print("                /     /                       \\\\")
print("              /      /                         \\ \\")
print("             /     /                            \\  \\")
print("           /     /             _----_            \\   \\")
print("          /     /           _-~      ~-_         |   |")
print("         (      (        _-~    _--_    ~-_     _/   |")
print(f"          \\      ~-____-~    _-~    ~-_    ~-_-~    /")
print(f"            ~-_           _-~          ~-_       _-~")
print(f"               ~--______-~                ~-___-~")
print("             ╔═════════════════════════════╗")
print("             ║    Structs en Python 󰌠      ║")
print("             ║-----------------------------║")
print(f"             ║ Tiempo de acceso: {elapsed_time:.2f} ms ║")
print("             ╚═════════════════════════════╝")

File: Structs/struct.rs
use std::time::Instant;

struct Persona {
    nombre: String,
    edad: u32,
    id: u32,
}

fn main() {
    // Creación del array
    let personas: Vec<Persona> = (0..1000000)
        .map(|i| Persona {
            nombre: "Josue".to_string(),
            edad: 25,
            id: i,
        })
        .collect();

    // Medición de tiempo de acceso
    let start_time = Instant::now();
    for persona in &personas {
        let _ = &persona.nombre;
        let _ = persona.edad;
        let _ = persona.id;
    }
    let elapsed_time = start_time.elapsed().as_millis();
    println!("            _______  ");
    println!("          _/       \\_");
    println!("         / |       | \\");
    println!("        /  |__   __|  \\");
    println!("       |__/((o| |o))\\__|");
    println!("       |      | |      |");
    println!("       |\\     |_|     /|");
    println!("       | \\           / |");
    println!("        \\| /  ___  \\ |/");
    println!("         \\ | / _ \\ | /");
    println!("          \\_________/");
    println!("           _|_____|_");
    println!("      ____|_________|____");
    println!("     /                   \\"); 
    println!("╔═════════════════════════════╗");
    println!("║     Structs en Rust 🦀      ║");
    println!("║-----------------------------║");
    println!("║ Tiempo de compilación: {:.2}ms  ║", elapsed_time);
    println!("╚═════════════════════════════╝");
  
}
